### PR TITLE
Puts test message behind a captcha

### PR DESCRIPTION
**Pull request by dakota002**
Adds a Modal with recaptcha to prevent spamming of test email functionality

### CODE REVIEW

File: app/routes/user.email._index/EmailNotificationCard.tsx [hunk 1]
```diff
@@ -151,6 +154,51 @@ export default function EmailNotificationCard({
...
+        </ModalHeading>
+        <p id="modal-confirm-description">
+          This will send a generic test email to {recipient}. Do you wish to
+          continue?
+        </p>
+        <ReCAPTCHA
+          onChange={(value) => {
+            setRecaptchaValid(Boolean(value))
+          }}
+        />
```
Comment (lpsinger) [inline app/routes/user.email._index/EmailNotificationCard.tsx:175]: The ReCAPTCHA component probably has to be inside the form.

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/user.email._index/EmailNotificationCard.tsx (hunk 1 above)
Comment (dakota002) [reply app/routes/user.email._index/EmailNotificationCard.tsx]: Woops, fixed in latest commit

File: app/routes/user.email._index/EmailNotificationCard.tsx [hunk 2]
```diff
@@ -151,6 +154,51 @@ export default function EmailNotificationCard({
...
+              outline
+              data-close-modal
+              disabled={!(recaptchaValid && testFetcher.state === 'idle')}
+            >
+              <Icon.MailOutline
+                role="presentation"
+                className="bottom-aligned margin-right-05"
+              />
+              Send
+            </Button>
```
Comment (lpsinger) [inline app/routes/user.email._index/EmailNotificationCard.tsx:198]: Put these in a button group and fix the alignment:

Screenshot 2024-09-03 at 19 45 46

**Review by lpsinger — CHANGES_REQUESTED** [review]
If you send a test message once and then click Test Message a second time, the ReCAPTCHA shows as solved, but the Send button is disabled:

Screenshot 2024-09-03 at 19 46 58

Please fix.

**Review by lpsinger — APPROVED** [review]

**Pull request merged**